scala/sort/merge_sort.sc

```diff
@@ -0,0 +1,31 @@
+object MergeSort {
+  def apply(comparisonPredicate: (Int, Int) => Boolean)(seq: Seq[Int]): Seq[Int] = {
+    seq match {
+      case first +: second +: tail =>
+        val pivot = Math.round((seq.length - 1).toFloat / 2)
```

Comment:
Is this pivot to split the array into halves?
Why not `seq.length / 2`?

Reply:
The idea was that the upper index is `length - 1`. And I need to convert it to float before dividing it. If it's an odd number, the result needs to be rounded, which is what I'm doing

Reply:
Cool! I'm not even sure Scala allows you to divide integers though

Reply:
Yea, it does: https://www.scala-lang.org/api/2.12.3/scala/Int.html#/(x:Int):Int

Reply:
:thinking: ok cool, good catch

Reply:
Agree with @paulanggono about this one. Will merge after this is updated 👍